### DISCUSSION TITLE
Display of Chinese characters

### DISCUSSION THREAD

**Discussion by linusqzdeng** [kdbinsidebrains/plugin#87] [Q&A] [answered]
Hi smklimenko. Sorry for not replying your answer last time. I have tried the [example](https://code.kx.com/q/kb/unicode/) on official website, and it works in kdb console.

![image](https://github.com/kdbinsidebrains/plugin/assets/68272577/e7360464-ee23-4112-bba0-0c13ff1f6720)

However, the data I have is retrieved from mysql database using [odbc interface](https://code.kx.com/q/interfaces/q-client-for-odbc/). The columns that can not be properly displayed in the kdb console are `varchar(100)` data type in mysql. But it can be displayed in the terminal when I changed the code page to 936 (GBK encoding)

![image](https://github.com/kdbinsidebrains/plugin/assets/68272577/cb4665a8-ee33-4915-8a68-2e4eef3be0c6)
![image](https://github.com/kdbinsidebrains/plugin/assets/68272577/24fe7be5-067e-4bd8-a1d3-f7a1c5b947a3)
 
So I guess is the plugin does not support GBK encoding? I will attach the data file. Thanks a lot if you have time to take a look.
[industry.zip](https://github.com/kdbinsidebrains/plugin/files/13805089/industry.zip)

**Comment by smklimenko** [comment, marked as answer]
ok, so, you have another encoding for your data. Unfortunately, there is no way do change encoding for the console as the data is processed and converted inside connection but what I can do - set encoding for a connection instead that you can defined in an instance configuration. Something like that:
![image](https://github.com/kdbinsidebrains/plugin/assets/3913096/3fccebb1-7062-48a5-8bc6-ea9007ab9264)

In this case all text data received from the instance will be converted using the encoding but if you have one encoding for one data and another for another table that won't work. Actually, it won't work in any case as there is no way to identify encoding looking at a random text data.

Will the solution with a connection encoding work for you?

**Reply by linusqzdeng** [reply]
Sure! That would be great if you could add this functionality during the instance creation. Thanks for your instant reply again!

**Reply by smklimenko** [reply]
ok. it will take a day or two to be approved but jetBrains but if you wish, you can download and instance version 5.2.0 manually to try it: https://plugins.jetbrains.com/plugin/16746-kdbinsidebrains/versions/stable/461933

The option called codepage and as other options can be set in Settings, per Scope or per Instance, depends on what's better for your case.
